### PR TITLE
Add sign in feature

### PR DESCRIPTION
**Pull request by Fatpaher**
- Add monban and monban generator gems
- Add User model with validations
- Add Sign in Sign Out and Sign Up links in menu
- Add User migration to database
- Add users and sessions controllers
- Add sign up page

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -13,6 +13,8 @@ gem 'jquery-rails'
 gem 'kaminari'
 gem 'puma'
 gem 'haml'
+gem 'monban'
+gem 'monban-generators'
```
Comment (wildfiler) [inline Gemfile]: @Fatpaher Please put it to `development` group.

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<%= form_for @user do |form| %>
```
Comment (wildfiler) [inline app/views/users/new.html.erb]: @Fatpaher Please, convert it to haml.

File: spec/factories.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
+Monban.test_mode!
```
Comment (wildfiler) [inline spec/factories.rb]: @Fatpaher Why this line here? I don't think that this is related to factories. Move it to `rails_helper.rb` please

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+Monban.test_mode!
```
Comment (wildfiler) [inline spec/models/user_spec.rb]: @Fatpaher same here. It should be in `rails_helper.rb`

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+describe User do
+  describe 'validations' do
+    it { should validate_presence_of(:email) }
+    it { should validate_presence_of(:password_digest) }
+  end
+
+  describe 'when email format is invalid' do
+    it 'should be invalid' do
```
Comment (wildfiler) [inline spec/models/user_spec.rb:10]: @Fatpaher Try this:

``` ruby
it { should_not allow_value(%w[user@foo,com user_at_foo.org example.user@foo.]).for(:email) }
```

File: spec/models/user_spec.rb (hunk 4 above)
Comment (Fatpaher) [inline spec/models/user_spec.rb]: @wildfiler I remove all of them in my next PR, but will fix it here to

**Pull request merged**